streams/shared/src/main/scala/zio/stream/internal/ChannelExecutor.scala

```diff
@@ -194,22 +194,28 @@ private[zio] class ChannelExecutor[Env, InErr, InElem, InDone, OutErr, OutElem,
                     currentChannel = read.more(out)
                   } catch {
                     case e: Throwable =>
-                      currentChannel = read.done.onExit(Exit.die(e))
+                      currentChannel = read.onHalt(Cause.die(e))
                   }
                   null
                 },
                 onDone = { (exit: Exit[Any, Any]) =>
-                  val onExit: Exit[Any, Any] => Channel[Env] =
-                    (read.done.onExit _).asInstanceOf[Exit[Any, Any] => Channel[Env]]
-                  currentChannel = onExit(exit)
+                  currentChannel = exit match {
+                    case s: Exit.Success[Any] => read.onSuccess(s.value)
+                    case f: Exit.Failure[Any] => read.onHalt(f.cause)
+                  }
```

Comment:
Here and in line [197](https://github.com/zio/zio/pull/9791/files#diff-0bc06e8bb285545c6a1ea6aa57d410c0e0fb297f9fbed25d04a71f496ee733faR197) were the usages of the `Fold.K` which were instantiated in each `Read` 

I remove these `Fold.K` instances in `Read` (See https://github.com/zio/zio/pull/9791/files#r2044495755), so I need to adapt this code